src/helm/benchmark/run_specs.py

```diff
@@ -2442,129 +2478,27 @@ def get_cleva_spec(
-        )
-        metric_specs = get_exact_match_metric_specs() + get_cleva_generative_harms_metric_specs()
-    elif task in ["reasoning_primitive"]:
-        max_tokens: int = 100 if subtask in ["variable_sub"] else 10
-
-        adapter_spec = AdapterSpec(
-            method=ADAPT_GENERATION,
-            instructions=format_instructions(prompt_setting.instructions),
-            input_prefix="",
-            input_suffix="",
-            output_prefix="",
-            max_train_instances=5,
-            num_outputs=1,
-            max_tokens=max_tokens,
-            temperature=1,
-            stop_sequences=["\n"],
-        )
-        metric_specs = get_exact_match_metric_specs() + get_cleva_generative_harms_metric_specs()
-    elif task in ["data_to_text_generation"]:
-        adapter_spec = AdapterSpec(
-            method=ADAPT_GENERATION,
-            instructions=format_instructions(prompt_setting.instructions),
-            input_prefix="",
-            output_prefix="",
-            max_train_instances=5,
-            num_outputs=1,
-            max_tokens=200,
-            temperature=1,
-        )
-        metric_specs = get_basic_metric_specs(["chinese_bleu_1"]) + get_cleva_generative_harms_metric_specs()
-    # elif task in ["mathematical_reasoning"]:
-    #     adapter_spec = get_generation_adapter_spec(
-    #         instructions=prompt_setting.instructions,
-    #         input_noun=prompt_setting.input_noun,
-    #         newline_after_input_noun=prompt_setting.newline_after_input_noun,
-    #         output_noun=prompt_setting.output_noun,
-    #         newline_after_output_noun=prompt_setting.newline_after_output_noun,
-    #         max_train_instances=5,  # limited by the context length
-    #         max_tokens=200,
-    #     )
-    #     metric_specs = get_basic_metric_specs(["chinese_math_result_match"])\
-    #          + get_cleva_generative_harms_metric_specs()
-    elif task in ["language_modeling"]:
-        adapter_spec = get_language_modeling_adapter_spec()
-        metric_specs = get_basic_metric_specs([])
-    else:
-        raise ValueError(f"The specified task '{task}' is not supported")
+        metric_specs = get_cleva_generative_harms_metric_specs()
+        if task in [
+            "opinion_mining",
+            "paraphrase_generation",
+            "closed_book_question_answering",
+            "conceptual_generalization",
+            "translation",
+            "mathematical_calculation",
+            "inductive_reasoning",
+            "reasoning_primitive",
+            "subject_knowledge",
+        ]:
+            metric_specs += get_cleva_generative_task_metric_spec(task, subtask)
+        elif task in ["closed_book_question_answering"]:
+            metric_specs += get_basic_metric_specs(["chinese_rouge_2"])
+        elif task in ["pinyin_transliteration", "dialogue_generation", "data_to_text_generation"]:
+            metric_specs += get_basic_metric_specs(["chinese_bleu_1"])
```

Comment:
I also think that we do not need that much "if-else" to get the accuracy metrics for different scenarios. We may avoid this by using `partial` in `get_cleva_generative_task_metric_spec` or passing `kwargs` to `get_cleva_generative_task_metric_spec`.

Reply:
The new version uses `partial` and puts the metrics of all the generation scenarios into `get_cleva_generative_task_metric_spec`.